src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs

```diff
@@ -776,19 +774,25 @@ public Bindings(ModelLoadContext ctx, TextLoader parent)
                     bool isKey = ctx.Reader.ReadBoolByte();
                     if (isKey)
                     {
+                        ulong count;
                         Contracts.CheckDecode(KeyType.IsValidDataKind(kind));
-
-                        bool isContig = ctx.Reader.ReadBoolByte();
-                        ulong min = ctx.Reader.ReadUInt64();
-                        Contracts.CheckDecode(min >= 0);
-                        int count = ctx.Reader.ReadInt32();
+                        if (ctx.Header.ModelVerWritten < 0x0001000C)
+                        {
+                            bool isContig = ctx.Reader.ReadBoolByte();
```

Comment:
>isContig [](start = 33, length = 8)

You can check that this is true and throw if it isn't.

Reply:
I this case I translate the old KeyType into a new one that is a superset of the old one. 

---
In reply to: [249206893](https://github.com/dotnet/machinelearning/pull/2146#discussion_r249206893) [](ancestors = 249206893)

Reply:
I think we should throw as @yaeldekel suggests. The reason is that we are now suddenly loading things as a very, very different type now. If anyone *happened* to be making use of this functionality (which, frankly, I doubt, but let's just suppose they were) your suggested change does not strike me as necessarily one that they would welcome. Indeed in some cases it would not even be possible... `U1[5000-5002]` used to be legal and sensible, now if it maps to `U1[0-50002]`, what then? So I agree with @yaeldekel that throwing here is best policy.

Reply:
Ok sounds good! I will update this then.

---
In reply to: [249908025](https://github.com/dotnet/machinelearning/pull/2146#discussion_r249908025) [](ancestors = 249908025)